optimizely/odp/zaius_graphql_api_manager.py

```diff
@@ -0,0 +1,203 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import json
+from sys import version_info
+from typing import Optional
+
+import requests
+
+from optimizely import logger as optimizely_logger
+from optimizely.helpers.enums import Errors
+
+if version_info < (3, 8):
+    from typing_extensions import Final
+else:
+    from typing import Final  # type: ignore
+
+"""
+ ODP GraphQL API
+ - https://api.zaius.com/v3/graphql
+ - test ODP public API key = "W4WzcEs-ABgXorzY7h1LCQ"
+
+
+ [GraphQL Request]
+
+ # fetch info with fs_user_id for ["has_email", "has_email_opted_in", "push_on_sale"] segments
+ curl -i -H 'Content-Type: application/json' -H 'x-api-key: W4WzcEs-ABgXorzY7h1LCQ' -X POST -d
+ '{"query":"query {customer(fs_user_id: \"tester-101\") {audiences(subset:[\"has_email\",
+ \"has_email_opted_in\", \"push_on_sale\"]) {edges {node {name state}}}}}"}' https://api.zaius.com/v3/graphql
+ # fetch info with vuid for ["has_email", "has_email_opted_in", "push_on_sale"] segments
+ curl -i -H 'Content-Type: application/json' -H 'x-api-key: W4WzcEs-ABgXorzY7h1LCQ' -X POST -d
+ '{"query":"query {customer(vuid: \"d66a9d81923d4d2f99d8f64338976322\") {audiences(subset:[\"has_email\",
+ \"has_email_opted_in\", \"push_on_sale\"]) {edges {node {name state}}}}}"}' https://api.zaius.com/v3/graphql
+
+    query MyQuery {
+    customer(vuid: "d66a9d81923d4d2f99d8f64338976322") {
+     audiences(subset:["has_email", "has_email_opted_in", "push_on_sale"]) {
+       edges {
+         node {
+           name
+           state
+         }
+       }
+     }
+    }
+    }
+
+
+    [GraphQL Response]
+    {
+    "data": {
+     "customer": {
+       "audiences": {
+         "edges": [
+           {
+             "node": {
+               "name": "has_email",
+               "state": "qualified",
+             }
+           },
+           {
+             "node": {
+               "name": "has_email_opted_in",
+               "state": "qualified",
+             }
+           },
+            ...
+         ]
+       }
+     }
+    }
+    }
+
+    [GraphQL Error Response]
+    {
+    "errors": [
+     {
+       "message": "Exception while fetching data (/customer) : java.lang.RuntimeException:
+       could not resolve _fs_user_id = asdsdaddddd",
+       "locations": [
+         {
+           "line": 2,
+           "column": 3
+         }
+       ],
+       "path": [
+         "customer"
+       ],
+       "extensions": {
+         "classification": "InvalidIdentifierException"
+       }
+     }
+    ],
+    "data": {
+     "customer": null
+    }
+    }
+"""
+
+
+REQUEST_TIMEOUT: Final = 10
```

Comment:
I don't see why not. Except that it will show reference to ConfigManager.

Reply:
Added a new class EventDispatchTimeout in enums that deals with any event dispatching requests.
This way we keep REQUEST_TIMEOUT in enums.ConfigManager for fetching datafile used specifically in config manager. @andrewleap-optimizely What do you think?

Reply:
If there were more configuration settings that we wanted to group together, I think that would be a good solution. However in this case, I think it makes more sense to just change the `REQUEST_TIMEOUT` global into a class var.

Reply:
After chatting w jae, decided to add an enum class for each instance: for events, odp graphql API configs, odp rest API configs. This could come handy later.